Durak/game_classes.py

```diff
@@ -69,9 +69,67 @@ def deal_one(self) -> Card:
     def deal_hand(self, hand_size: int) -> list:
         hand = [self.cards.pop(0) for i in range(hand_size)]        
         return hand
+
+class Player:
+    def __init__(self, name: str):
+        self.name = name
+        self.hand = []
+
+    def __str__(self):
+        return self.name
+    
+    def deal_hand(self, hand: list):
+        self.hand = hand
+
+class Challenge:
+    def __init__(self, offense: Card, attacker: Player):
+        self.offense = offense
+        self.attacker = attacker
+        self.defended = 0
+
+    def __str__(self):
+        string = f"{str(self.attacker)} attacks with {str(self.offense)}"
+        if(self.defended == 1):
+            string += f"; defended with {str(self.defense)}"
+        return string
+
+    def defend(self, defense: Card):
+        if(defense > self.offense):
+            self.defense = defense
+            self.defended = 0
+            return 1
+        else:
+            return 0
     
 class GameField:
-    pass
+    def __init__(self, deck: Deck, players: list):
+        self.deck = deck
+        self.players = players
+        self.challenges = []
+        self.open_attack = 0
 
-class Player:
-    pass
\ No newline at end of file
+    def set_defender(self, defender: Player):
+        self.defender = defender
+
+    def attack(self, card: Card, attacker: Player):
+        if(card.value == "Joker"):
+            return 0
+        
+        can_play = 0
+        if(len(self.challenges) == 0):
+            can_play = 1
+        else:
+            for n in self.challenges:
```

Comment:
Insyead of using 0 and 1 you can use True and False instead and it makes the code easier since you can write: "if n.defended:" instead of "if n.defended == 1:"